xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -499,6 +504,7 @@ func (b *clusterImplBalancer) run() {
 						Picker:            b.newPicker(dc),
 					})
 				}
+				b.cfgUpdateDone <- struct{}{}
```

Comment:
IMO, we should close this channel instead. This would ensure that the code panics if the channel is accidentally used multiple times. Presently it will just hang forever and may be harder to debug.

Reply:
You are right that the channel needs to be closed. But this channel cannot be a field on the LB policy struct. This channel needs to be created and closed once per-config update. See my other comment with details about the complete fix that I think should be made.